### PR TITLE
Non rotation icon  profile settings redesign

### PR DESCRIPTION
**Pull request by nnngrach**
[issue](https://github.com/osmandapp/OsmAnd-iOS/issues/3553)

Before: 
many layout bugs on mac and ipad versions
Screenshot 2024-09-17 at 15 01 35

After: 
layout bug fixed. New components with one line and horizontal scroll. Two new settings at bottom.
Screenshot 2024-09-08 at 22 55 01

Video 1

https://github.com/user-attachments/assets/b83f4b0d-e09e-45ce-adb3-e5b2251bc506

Video 2

https://github.com/user-attachments/assets/bc8ffe07-8d2a-46e8-b63f-f08ee1fdc964

Video 3

https://github.com/user-attachments/assets/5adc11af-d38f-4f5e-904c-694f2307757b

Screenshot 2024-09-08 at 23 15 48 copy

Before:
All colors - bug with only one line always showing.
Screenshot 2024-09-17 at 15 04 47

After: 
Video 4 - All colors shows several lines

https://github.com/user-attachments/assets/4e6d299c-95e2-4d47-9715-9b36e6216ecd

### CODE REVIEW

File: Sources/Controllers/Cells/OAIconsPaletteCell.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+//
+//  OAIconsPaletteCell.h
+//  OsmAnd
+//
+//  Created by Max Kojin on 20.08.2024.
+//  Copyright © 2023 OsmAnd. All rights reserved.
+//
+
+#import "OACollectionSingleLineTableViewCell.h"
+#import "OASuperViewController.h"
```
Comment (tigrim) [inline Sources/Controllers/Cells/OAIconsPaletteCell.h:10]: @class OASuperViewController; ?

File: Sources/Controllers/Map/Layers/OAMyPositionLayer.mm [hunk 2]
```diff
@@ -97,6 +83,7 @@ - (void) updateLocation:(OsmAnd::PointI)target31 animationDuration:(float)animat
 @implementation OAMarkerCollection
 {
     OAMapRendererView *_mapView;
+    BOOL _chachedHeading;
```
Comment (tigrim) [inline Sources/Controllers/Map/Layers/OAMyPositionLayer.mm:86]: cached?

File: Sources/Controllers/Map/Layers/OAMyPositionLayer.mm [hunk 3]
```diff
@@ -833,6 +689,16 @@ - (void) updateMyLocationCourseProvider
     
         OAApplicationMode *currentMode = [OAAppSettings sharedManager].applicationMode.get;
         
+        for (OAApplicationMode *mode in _modeMarkers.keyEnumerator)
+        {
+            OAMarkerCollection *c = [_modeMarkers objectForKey:mode];
+            if (mode != currentMode)
```
Comment (tigrim) [inline Sources/Controllers/Map/Layers/OAMyPositionLayer.mm:695]: && c

File: Sources/Controllers/Map/Layers/OAMyPositionLayerState.h [hunk 4]
```diff
@@ -0,0 +1,15 @@
+//
+//  OAMyPositionLayerState.h
+//  OsmAnd
+//
+//  Created by Max Kojin on 08/09/24.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+typedef enum {
```
Comment (tigrim) [inline Sources/Controllers/Map/Layers/OAMyPositionLayerState.h:9]: NS_ENUM

File: Sources/Controllers/Settings/OAProfileAppearanceViewController.mm [hunk 5]
```diff
@@ -388,6 +450,52 @@ - (void) generateData
                @"ic_action_kayak",
                @"ic_action_motorboat",
                @"ic_action_light_aircraft"];
+    
+    _profileIconCollectionHandler =  [[IconCollectionHandler alloc] initWithData:@[_icons] collectionView:nil];
```
Comment (tigrim) [inline Sources/Controllers/Settings/OAProfileAppearanceViewController.mm:454]: remove space

File: Sources/Controllers/Settings/ProfileAppearanceLocationRadiusViewController.swift [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+//  ProfileAppearanceLocationRadiusViewController.swift
+//  OsmAnd Maps
+//
+//  Created by Max Kojin on 04/09/24.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+import UIKit
+
+@objc(OAProfileAppearanceLocationRadiusUpdatable)
```
Comment (tigrim) [inline Sources/Controllers/Settings/ProfileAppearanceLocationRadiusViewController.swift:11]: There's no need to specify a name for Objective-C

File: Sources/Controllers/Settings/ProfileAppearanceLocationRadiusViewController.swift [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+protocol ProfileAppearanceLocationRadiusUpdatable: AnyObject {
+    func onLocationRadiusUpdated(newValue: Int)
+}
+
+@objc(OAProfileAppearanceLocationRadiusViewController)
```
Comment (tigrim) [inline Sources/Controllers/Settings/ProfileAppearanceLocationRadiusViewController.swift:16]: up

File: Sources/Controllers/Settings/ProfileAppearanceLocationRadiusViewController.swift [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+            return cell
+        }
+        return nil
+    }
+    
+    override func onRowSelected(_ indexPath: IndexPath) {
+        let item = tableData.item(for: indexPath)
+        let selectedRawValue = item.integer(forKey: Self.rawValueKey)
+        if let delegate {
+            delegate.onLocationRadiusUpdated(newValue: selectedRawValue)
```
Comment (tigrim) [inline Sources/Controllers/Settings/ProfileAppearanceLocationRadiusViewController.swift:78]: delegate?.

File: Sources/Controllers/Settings/ProfileAppearanceViewAngleViewController.swift [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+//
+
+import UIKit
+
+@objc(OAProfileAppearanceViewAngleUpdatable)
+protocol ProfileAppearanceViewAngleUpdatable: AnyObject {
+    func onViewAngleUpdated(newValue: Int)
+}
+
+@objc(OAProfileAppearanceViewAngleViewController)
```
Comment (tigrim) [inline Sources/Controllers/Settings/ProfileAppearanceViewAngleViewController.swift:16]: up

File: Sources/Controllers/Settings/ProfileAppearanceViewAngleViewController.swift [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+            return cell
+        }
+        return nil
+    }
+    
+    override func onRowSelected(_ indexPath: IndexPath) {
+        let item = tableData.item(for: indexPath)
+        let selectedRawValue = item.integer(forKey: Self.rawValueKey)
+        if let delegate {
+            delegate.onViewAngleUpdated(newValue: selectedRawValue)
```
Comment (tigrim) [inline Sources/Controllers/Settings/ProfileAppearanceViewAngleViewController.swift:78]: up

File: Sources/Controllers/TargetMenu/GPX/TrackMenu/Appearance/OAColorCollectionViewController.m [hunk 11]
```diff
@@ -131,7 +146,7 @@ - (void)viewWillAppear:(BOOL)animated
 
 - (NSString *)getTitle
 {
-    return OALocalizedString(@"shared_string_all_colors");
+    return (_customTitle && _customTitle.length > 0) ? _customTitle : OALocalizedString(@"shared_string_all_colors");
```
Comment (tigrim) [inline Sources/Controllers/TargetMenu/GPX/TrackMenu/Appearance/OAColorCollectionViewController.m:149]: just _customTitle.length > 0

File: Sources/Controllers/Settings/MarkerDisplayOption.swift [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+//  OsmAnd Maps
+//
+//  Created by Max Kojin on 04/09/24.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+import UIKit
+
+@objcMembers
+final class MarkerDisplayOption: NSObject {
```
Comment (tigrim) [inline Sources/Controllers/Settings/MarkerDisplayOption.swift:12]:
```

final class MarkerDisplayOptionWrapper: NSObject {

    static func getNameFor(type: MarkerDisplayOption) -> String {
        type.nameId
    }
}

enum MarkerDisplayOption: Int32, CaseIterable {
    case off, resting, navigation, restingNavigation

    // MARK: - Properties

    var nameId: String {
        switch self {
        case .off:
            return "shared_string_off"
        case .resting:
            return "resting_position"
        case .navigation:
            return "navigation_position"
        case .restingNavigation:
            return "resting_navigation_position"
        }
    }

    var markerStates: [EOAMarkerState] {
        switch self {
        case .off:
            return [OAMarkerStateNone]
        case .resting:
            return [OAMarkerStateStay]
        case .navigation:
            return [OAMarkerStateMove]
        case .restingNavigation:
            return [OAMarkerStateMove, OAMarkerStateStay]
        }
    }

    // MARK: - Methods

    static func value(by index: Int32) -> MarkerDisplayOption {
        MarkerDisplayOption(rawValue: index) ?? .off
    }

    func name() -> String {
        localizedString(nameId)
    }

    func isVisible(markerState: EOAMarkerState) -> Bool {
        self != .off && markerStates.contains(markerState)
    }
}
```

**Comment by nnngrach** [comment]
fixed on movement behavior

https://github.com/user-attachments/assets/97ae4a6a-5ccc-47fb-9705-90708cbc0611

File: Sources/Controllers/Cells/Handlers/IconCollectionHandler.swift [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+    
+    func setIconSize(size: CGFloat) {
+        iconSize = size
+    }
+    
+    func getIconSize() -> CGFloat{
+        iconSize ?? 30
+    }
+    
+    override func getCellIdentifier() -> String {
```
[the remark this answers is not in the document]
Comment (nnngrach) [reply Sources/Controllers/Cells/Handlers/IconCollectionHandler.swift:49]: It calls by parent class
Screenshot 2024-09-09 at 18 56 12

File: Sources/Controllers/Cells/Handlers/IconCollectionHandler.swift [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+            cell.iconImageView.tintColor = regularIconColor
+            cell.backView.layer.borderColor = UIColor.clear.cgColor
+            cell.backView.layer.borderWidth = 0
+        }
+        if cornerRadius != -1 {
+            cell.iconView.layer.cornerRadius = cornerRadius
+            cell.backView.layer.cornerRadius = cornerRadius
+        } else {
+            cell.iconView.layer.cornerRadius = cell.iconView.frame.size.height/2;
+            cell.backView.layer.cornerRadius = cell.backView.frame.size.height/2;
```
[the remark this answers is not in the document]
Comment (nnngrach) [reply Sources/Controllers/Cells/Handlers/IconCollectionHandler.swift:107]: backView is bigger that iconView. 
Screenshot 2024-09-09 at 19 38 31

**Comment by nnngrach** [comment]
Fixed color picker bug - not responding in mac os

https://github.com/user-attachments/assets/99905dd8-efd0-4ce2-9fc8-4d980c5b5c7e

https://github.com/user-attachments/assets/ffb3ddbd-8bfb-4928-b9ca-6a8bd517eb33

File: Sources/Controllers/Cells/Handlers/OAColorCollectionHandler.m [hunk 15]
```diff
@@ -356,12 +362,14 @@ - (void)onCollectionItemSelected:(NSIndexPath *)indexPath {
 
 - (void)selectColorItem:(OAColorItem *)colorItem
 {
+    NSIndexPath *prevSelectedColorIndex = [self getSelectedIndexPath];
     NSIndexPath *selectedIndex = [NSIndexPath indexPathForRow:[_data[0] indexOfObject:colorItem] inSection:0];
     [self setSelectedIndexPath:selectedIndex];
+    [[self getCollectionView]  reloadItemsAtIndexPaths:@[prevSelectedColorIndex, selectedIndex]];
```
Comment (tigrim) [inline Sources/Controllers/Cells/Handlers/OAColorCollectionHandler.m:368]: remove space

File: Sources/Controllers/TargetMenu/PointEditing/OABaseEditorViewController.mm [hunk 16]
```diff
@@ -424,6 +393,9 @@ - (void)setupColors
 {
     _selectedColorItem = [_appearanceCollection getColorItemWithValue:[self.editColor toARGBNumber]];
     _sortedColorItems = [NSMutableArray arrayWithArray:[_appearanceCollection getAvailableColorsSortingByLastUsed]];
+    _colorCollectionHandler =  [[OAColorCollectionHandler alloc] initWithData:@[_sortedColorItems] collectionView:nil];
```
Comment (tigrim) [inline Sources/Controllers/TargetMenu/PointEditing/OABaseEditorViewController.mm:396]: remove space

File: Sources/Controllers/Cells/Handlers/IconCollectionHandler.swift [hunk 17]
```diff
@@ -114,7 +114,6 @@ final class IconCollectionHandler: OABaseCollectionHandler, OAIconCollectionDele
         var vc: OAColorCollectionViewController?
         if !iconImagesData.isEmpty {
```
Comment (tigrim) [inline Sources/Controllers/Cells/Handlers/IconCollectionHandler.swift:115]: use ternary operation

**Pull request merged**